jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListPresenter.java

```diff
@@ -130,6 +134,23 @@ protected DataSetReadyCallback getDataSetReadyCallback(final Integer startRange,
                 .withEmptyResultsCallback(() -> setEmptyResults());
     }
 
+    private Set<Long> findDuplicateId(LinkedList<TaskSummary> taskSummaries) {
+        Set<Long> id = new HashSet<>();
+        return taskSummaries.stream().filter(ts -> !id.add(ts.getId())).map(TaskSummary::getId).collect(Collectors.toSet());
+    }
+
+    protected void removeDuplicateTaskSummaryBySLAStatus(LinkedList<TaskSummary> taskSummaries) {
+        Set<Long> duplicateTaskSummaries = findDuplicateId(taskSummaries);
+        if (!duplicateTaskSummaries.isEmpty()) {
+            for (int i = 0; i < taskSummaries.size(); i++) {
+                Integer slaPending = new Integer(ProcessInstance.SLA_PENDING);
```

Comment:
This variable definition can be outside the loop, there is no need to create it every time.

Reply:
yes you are right